libvast/vast/concept/parseable/vast/schema.hpp

```diff
@@ -18,53 +18,183 @@
 #include "vast/concept/parseable/string.hpp"
 #include "vast/concept/parseable/vast/identifier.hpp"
 #include "vast/concept/parseable/vast/type.hpp"
+#include "vast/detail/overload.hpp"
+#include "vast/error.hpp"
+#include "vast/logger.hpp"
 #include "vast/schema.hpp"
 #include "vast/type.hpp"
 
 namespace vast {
 
-struct schema_parser : parser<schema_parser> {
+struct shared_schema_parser : parser<shared_schema_parser> {
```

Comment:
Please add documentation here, explaining how this type relates to `schema_parser`.